### PR TITLE
feat: Combine workstreams

### PR DESCRIPTION
**Pull request by nicain**

### CODE REVIEW

File: terraform/webhook/bigquery.py [hunk 1]
```diff
@@ -50,8 +52,9 @@ def write_summarization_to_table(
       summary (str): the text summary of the document
       summary_uri (str): the Storage URI of the summary TXT document
       timestamp (datetime): when the processing occurred
+      credentials (Credentials): TODO(nicain)
     """
-    client = bigquery.Client()
+    client = bigquery.Client(credentials=credentials)
```
Comment (telpirion) [inline terraform/webhook/bigquery.py:57]: Do we need to pass in credentials when we instantiate the BQ client? The prototype worked without this step.

File: terraform/webhook/bigquery_test.py [hunk 2]
```diff
@@ -15,27 +15,29 @@
...
+        dataset_id=_DATASET_ID,
+        table_id=_TABLE_ID,
         bucket="gs://fake_bucket",
         filename="fake.pdf",
         complete_text="This is fake text of an academic paper",
         complete_text_uri="gs://fake_bucket/texts/fake.txt",
         summary="Some fake text summary",
         summary_uri="gs://fake_bucket/fake-summary.txt",
         timestamp=datetime.datetime.now(),
+        credentials=_CREDENTIALS,
```
Comment (telpirion) [inline terraform/webhook/bigquery_test.py:40]: Same as previous with regards to credentials.

File: terraform/webhook/services_test.py [hunk 3]
```diff
@@ -18,19 +18,21 @@
...
+from google.auth import default
+
+_PROJECT_ID = os.environ["PROJECT_ID"]
+_BUCKET_NAME = os.environ["BUCKET"]
+_OUTPUT_BUCKET = f'{_PROJECT_ID}_output'
+_DATASET_ID = "summary_dataset"
+_TABLE_ID = "summary_table"
+_FILE_NAME = '9404001v1.pdf'
+_MODEL_NAME = 'text-bison@001'
+_CREDENTIALS, _ = default(scopes=['https://www.googleapis.com/auth/cloud-platform'])
```
Comment (telpirion) [inline terraform/webhook/services_test.py:35]: Same as previous, w/r/t credentials

File: terraform/webhook/bigquery.py (hunk 1 above)
Comment (nicain) [reply terraform/webhook/bigquery.py]: Removed.

File: terraform/webhook/bigquery_test.py (hunk 2 above)
Comment (nicain) [reply terraform/webhook/bigquery_test.py]: Done.

File: terraform/webhook/main.py [hunk 4]
```diff
@@ -76,49 +52,124 @@ def summarize_text(text: str, parameters: None | dict[str, int | float] = None)
...
+        **final_parameters,
+    )
     print(f"Response from Model: {response.text}")
 
     return response.text
 
 
-@functions_framework.cloud_event
-def entrypoint(request):
+def entrypoint(request: object) -> dict[str, str]:
```
Comment (telpirion) [inline terraform/webhook/main.py:72]: question: do we no longer need the`functions_framework.cloud_event` decorator?

**Review by telpirion — APPROVED** [review]

File: terraform/webhook/main.py (hunk 4 above)
Comment (nicain) [reply terraform/webhook/main.py:72]: Yup, no longer necessary.

**Pull request merged**